service/init.go

```diff
@@ -141,91 +125,9 @@ func (amf *AMF) Initialize(ctx ctxt.Context, c *cli.Command) error {
 
 	factory.AmfConfig.CfgLocation = absPath
 
-	if os.Getenv("MANAGED_BY_CONFIG_POD") == "true" {
-		factory.AmfConfig.Configuration.ServedGumaiList = nil
-		factory.AmfConfig.Configuration.SupportTAIList = nil
-		factory.AmfConfig.Configuration.PlmnSupportList = nil
-		logger.InitLog.Infoln("Reading Amf related configuration from ROC")
-		go manageGrpcClient(ctx, factory.AmfConfig.Configuration.WebuiUri, amf)
-	} else {
-		go func() {
-			logger.GrpcLog.Infoln("reading Amf Configuration from Helm")
-			// sending true to the channel for sending NFRegistration to NRF
-			RocUpdateConfigChannel <- true
-		}()
-	}
-
 	return nil
 }
 
-// manageGrpcClient connects the config pod GRPC server and subscribes the config changes.
-// Then it updates AMF configuration.
-func manageGrpcClient(ctx ctxt.Context, webuiUri string, amf *AMF) {
-	var configChannel chan *protos.NetworkSliceResponse
-	var client grpcClient.ConfClient
-	var stream protos.ConfigService_NetworkSliceSubscribeClient
-	var err error
-	count := 0
-	for {
-		if client != nil {
-			if client.CheckGrpcConnectivity() != "READY" {
-				time.Sleep(time.Second * 30)
-				count++
-				if count > 5 {
-					err = client.GetConfigClientConn().Close()
-					if err != nil {
-						logger.InitLog.Infof("failing ConfigClient is not closed properly: %+v", err)
-					}
-					client = nil
-					count = 0
-				}
-				logger.InitLog.Infoln("checking the connectivity readiness")
-				continue
-			}
-
-			if stream == nil {
-				stream, err = client.SubscribeToConfigServer()
-				if err != nil {
-					logger.InitLog.Infof("failing SubscribeToConfigServer: %+v", err)
-					continue
-				}
-			}
-
-			if configChannel == nil {
-				configChannel = client.PublishOnConfigChange(true, stream)
-				logger.InitLog.Infoln("PublishOnConfigChange is triggered")
-				go amf.UpdateConfig(ctx, configChannel)
-				logger.InitLog.Infoln("AMF updateConfig is triggered")
-			}
-
-			time.Sleep(time.Second * 5) // Fixes (avoids) 100% CPU utilization
-		} else {
-			client, err = grpcClient.ConnectToConfigServer(webuiUri)
-			stream = nil
-			configChannel = nil
-			logger.InitLog.Infoln("connecting to config server")
-			if err != nil {
-				logger.InitLog.Errorf("%+v", err)
-			}
-			continue
-		}
-	}
-}
-
-func (amf *AMF) WatchConfig() {
-	viper.WatchConfig()
-	viper.OnConfigChange(func(e fsnotify.Event) {
-		logger.AppLog.Infoln("config file changed:", e.Name)
-		if err := factory.UpdateConfig(factory.AmfConfig.CfgLocation); err != nil {
-			logger.AppLog.Errorln("error in loading updated configuration")
-		} else {
-			self := context.AMF_Self()
-			util.InitAmfContext(self)
-			logger.AppLog.Infoln("successfully updated configuration")
-		}
-	})
-}
```

Comment:
Given that this function is removed, I suspect that the use of `viper` becomes obsolete. So, should we remove the following code from the `Initialize` function above?
```
	if _, err := os.Stat(absPath); err == nil {
		viper.SetConfigFile(absPath)
		viper.SetConfigType("yaml")
		err = viper.ReadInConfig() // Find and read the config file
		if err != nil {            // Handle errors reading the config file
			return err
		}
	} else if os.IsNotExist(err) {
		logger.AppLog.Errorln("file %s does not exists", absPath)
		return err
	}
```